addons/purchase_stock/static/tests/tours/purchase_catalog_suggest.js

```diff
@@ -0,0 +1,147 @@
+import { registry } from "@web/core/registry";
+import {
+    freezeDateTime,
+    selectPOVendor,
+    selectPOWarehouse,
+    goToCatalogFromPO,
+    goToPOFromCatalog,
+    toggleSuggest,
+    setSuggestParameters,
+} from "./tour_helpers";
+
+registry.category("web_tour.tours").add("test_purchase_catalog_suggest_search", {
+    steps: () => [
+        ...freezeDateTime("2021-01-14 09:12:15"), // Same date as python @freeze decorator
+        { trigger: ".o_purchase_order" },
+        {
+            content: "Create a New PO",
+            trigger: ".o_list_button_add",
+            run: "click",
+        },
+        ...selectPOVendor("Julia Agrolait"),
+        ...goToCatalogFromPO(),
+        toggleSuggest(),
+        { trigger: "button[name='toggle_suggest_catalog'].fa-toggle-off" },
```

Comment:
Can be part of the helper method (if you want to make the helper method still work for toggle on/toggle off, you can do two methods `toggleSuggestOn´/`toggleSuggestOff` or pass on/off as a parameter for example)